File: WebCrawlerAKKA/src/main/scala/com.en_workshop.webcrawlerakka.akka.requests/scala/MessageResponse.scala
package com.en_workshop.webcrawlerakka.akka.requests.scala

import com.en_workshop.webcrawlerakka.entities.Link

abstract class MessageResponse(messageRequest: MessageRequest, id: Long)

case class DownloadUrlResponse(downloadUrlRequest: DownloadUrlRequest) extends MessageResponse(downloadUrlRequest, System.currentTimeMillis())
case class NextLinkResponse(nextLinkRequest: NextLinkRequest, webUrl: Link) extends MessageResponse(nextLinkRequest, System.currentTimeMillis())

File: WebCrawlerAKKA/src/main/java/com/en_workshop/webcrawlerakka/akka/requests/MessageResponse.java
package com.en_workshop.webcrawlerakka.akka.requests;

/**
 * @author Radu Ciumag
 */
public abstract class MessageResponse {

    private final long id;
    private final MessageRequest messageRequest;

    public MessageResponse(final long id, final MessageRequest messageRequest) {
        this.id = id;
        this.messageRequest = messageRequest;
    }

    public long getId() {
        return id;
    }

    public MessageRequest getMessageRequest() {
        return messageRequest;
    }
}
